packages/foreman/katello-certs-tools/katello-certs-tools.spec

```diff
@@ -2,40 +2,62 @@ Name:      katello-certs-tools
 Summary:   Katello SSL Key/Cert Tool
 Group:     Applications/Internet
 License:   GPLv2
-Version:   2.4.0
+Version:   2.5.0
 Release:   1%{?dist}
 URL:       https://github.com/katello/katello-certs-tools
-Source0:   https://codeload.github.com/Katello/%{name}/tar.gz/%{version}
+Source0:   https://codeload.github.com/Katello/katello-certs-tools/tar.gz/%{version}#/%{name}-%{version}.tar.gz
 BuildArch: noarch
 
-#BZ 1022017
-Requires: openssl >= 1.0.1e-16
-
+Requires: openssl
 Requires: rpm-build
 
 BuildRequires: docbook-utils
-BuildRequires: python
+BuildRequires: /usr/bin/pathfix.py
+%if 0%{?rhel} > 7
+BuildRequires: python3-devel
+%else
+BuildRequires: python-devel >= 2.6
+%endif
 
 %description
 This package contains tools to generate the SSL certificates required by
 Katello.
 
 %prep
 %setup -q
+%if 0%{?rhel} > 7
+pathfix.py -pni "%{__python3} %{py3_shbang_opts}" . katello-ssl-tool
+%else
+pathfix.py -pni "%{__python2} %{py2_shbang_opts}" . katello-ssl-tool
+%endif
 
 %build
 /usr/bin/docbook2man katello-ssl-tool.sgml
+%if 0%{?rhel} > 7
+%{__python3} setup.py build
+%else
 %{__python} setup.py build
+%endif
 
 %install
+%if 0%{?rhel} > 7
+%{__python3} setup.py install --skip-build --root $RPM_BUILD_ROOT
+chmod +x $RPM_BUILD_ROOT/%{python3_sitelib}/katello_certs_tools/katello_ssl_tool.py
```

Comment:
Is this executable part actually needed? It's main is:
```python
if __name__ == "__main__":                                                              
    sys.stderr.write('\nWARNING: intended to be wrapped by another executable\n'
                     '           calling program.\n')
    sys.exit(abs(main() or errnoSuccess))  
```

Reply:
Fair question, in the current design, this file has a shebang at the top.

Reply:
Let's figure out if we can drop this in a future version then :)